File: Project2/Model/Movie.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Project2.Model
{
    public class Movie
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid MovieId { get; set; }

        [Required]
        public string Title { get; set; }

        public int? ReleaseYear { get; set; }

        public string Genre { get; set; }

        public string Rating { get; set; }

        [Required]
        public decimal RentalPrice { get; set; }

        public int TotalCopies { get; set; }

        // Navigation property to RentalDetails
        public ICollection<RentalDetail> RentalDetails { get; set; }

    }
}

File: Project2/ProjectUI/ProjectUI/app/model/Movie.js
Ext.define('ProjectUI.model.Movie', {
    extend: 'ProjectUI.model.Base',
    idProperty: 'movieId',

    fields: [
        { name: 'movieId', type: 'string' },
        { name: 'title', type: 'string' },
        { name: 'releaseYear', type: 'int' },
        { name: 'genre', type: 'string' },
        { name: 'rating', type: 'string' },
        { name: 'rentalPrice', type: 'float' },
        { name: 'totalCopies', type: 'int' }
    ]
});
